src/rapriori.c

```diff
@@ -1,1131 +1,1135 @@
+}
+
+SEXP rapriori(SEXP x, SEXP y, SEXP dim, SEXP parms, SEXP control, SEXP app, SEXP itemInfo)
+{
+  ARparameter param;
+  int k;
+  int load, maxlen;
+  const char *target, *arem;
+  clock_t t;
+  INPUT in;
+  SEXP ans;
+  
+  param.verbose = *LOGICAL(GET_SLOT(control, install("verbose")));  /* flag for verbose */
+    /*
+     if (param.verbose) {
+     Rprintf("%s - %s\n", PRGNAME, DESCRIPTION);
+     Rprintf(VERSION);
+     Rprintf("\n");
+     }
+     */
+    
+    /* --- evaluate arguments --- */
+    param.supp = *REAL(GET_SLOT(parms, install("support")));       /* minimal support 's'*/
+    param.conf = *REAL(GET_SLOT(parms, install("confidence")));    /* minimal confidence 'c'*/ 
+    param.minval = *REAL(GET_SLOT(parms, install("minval")));      /* minimal evaluation measure value 'd'*/
+    param.filter = *REAL(GET_SLOT(control, install("filter")));    /* item usage filtering parameter 'u'*/
+    param.smax = *REAL(GET_SLOT(parms, install("smax")));          /* maximal support    'S' */
+    /* target type (sets/rules/h.edges) 't'*/  
+    target = translateChar(STRING_ELT(GET_SLOT(parms, install("target")), 0)); 
+    /* target = CHAR(STRING_ELT(GET_SLOT(parms, install("target")), 0)); */
+    param.target = targetcode(target);
+    
+    /* additional rule evaluation measure 'e'*/  
+    arem = translateChar(STRING_ELT(GET_SLOT(parms, install("arem")), 0)); 
+    /* arem = CHAR(STRING_ELT(GET_SLOT(parms, install("arem")), 0)); */ 
+    param.arem = aremcode(arem);
+    
+    param.minlen = *INTEGER(GET_SLOT(parms, install("minlen")));   /* minimal rule length 'm'*/
+    maxlen = param.maxlen = *INTEGER(GET_SLOT(parms, install("maxlen")));   /* maximal rule length 'n'*/  
+    param.maxtime = *REAL(GET_SLOT(parms, install("maxtime")));      /* maximal time allowed in seconds*/
+    param.sort = *INTEGER(GET_SLOT(control, install("sort")));     /* flag for item sorting and recoding 'q'*/
+    
+    param.rsdef = *LOGICAL(GET_SLOT(parms, install("originalSupport")));      /* rule support definition 'o'*/
+    param.tree = *LOGICAL(GET_SLOT(control, install("tree")));      /* flag for transaction tree 'h'*/
+    param.heap = *LOGICAL(GET_SLOT(control, install("heap")));      /* flag for heap sort vs. quick sort 'j'*/
+    param.aval = *LOGICAL(GET_SLOT(parms, install("aval")));        /* flag for add. eval. measure value 'v'*/
+    param.memopt = *LOGICAL(GET_SLOT(control, install("memopt")));  /* flag for memory usage optimization 'z'*/
+    param.ext = *LOGICAL(GET_SLOT(parms, install("ext")));          /* flag for extended support output 'x'*/
+    param.trans = 0;
+    
+    load = *LOGICAL(GET_SLOT(control, install("load")));            /* flag for loading transactions 'l'*/
+    switch (param.target) {             /* check and translate target type */
+    case 0: param.target = TT_SET;               break;
+    case 1: param.target = TT_MFSET;             break;
+    case 2: param.target = TT_GRSET;             break;

```

Comment:
/* Makh2018: Initializing TT_GRSET as a target parameter to find generator item sets*/